packages/infrastructure/src/category/types.ts

```diff
@@ -0,0 +1,26 @@
+import { type Category } from "@nimara/codegen/schema";
+import { type AsyncResult } from "@nimara/domain/objects/Result";
+
+import { type FetchOptions } from "#root/graphql/client";
+import { type Logger } from "#root/logging/types";
+
+export type SaleorCategoryServiceConfig = {
+  apiURI: string;
+  logger: Logger;
+};
+
+type WithFetchOptions = { options?: FetchOptions };
+
+type CollectionsIDsBySlugsOptions = {
+  slugs: string[];
+} & WithFetchOptions;
+
+export type GetCategoriesIDsBySlugsInfra = (
+  opts: CollectionsIDsBySlugsOptions,
+) => AsyncResult<{
+  results: Array<Category["id"]> | null;
+}>;
```

Comment:
No need for extra wrapper (results):
```suggestion
) => AsyncResult<Array<Category["id"]>> | null>;
```